src/Umbraco.Cms.Api.Management/CLAUDE.md

```diff
@@ -0,0 +1,622 @@
+├── Services/                      # Business logic (thin layer over Core.Services)
+├── Mapping/                       # ViewModel → domain model mappers
+├── Security/                      # Auth providers, sign-in manager, external logins
+├── OpenApi/                       # Swashbuckle filters (schema, operation, security)
+├── Routing/                       # Route configuration, SignalR hubs
+├── DependencyInjection/           # Service registration (55+ files)
+├── Middleware/                    # Preview, server events
+├── Configuration/                 # IOptions configurators
+├── Filters/                       # Action filters
+├── Serialization/                 # JSON converters
+└── OpenApi.json                   # Embedded OpenAPI spec (1.3MB)
+```
+
+### Dependencies
+- **Umbraco.Cms.Api.Common** - Shared API infrastructure (base controllers, OpenAPI config)
+- **Umbraco.Infrastructure** - Service implementations, data access
+- **Umbraco.PublishedCache.HybridCache** - Published content queries
+- **JsonPatch.Net** - JSON Patch (RFC 6902) support
+- **Swashbuckle.AspNetCore** - OpenAPI generation
+
+### Design Patterns
+1. **Controller-per-Operation** - Each endpoint is a separate controller class
+   - Example: `CreateDocumentController`, `UpdateDocumentController`, `DeleteDocumentController`
+   - Enables fine-grained authorization and operation-specific logic
+
+2. **Presentation Factory Pattern** - Factories convert domain models to ViewModels
+   - Example: `IDocumentEditingPresentationFactory` (src/Umbraco.Cms.Api.Management/Factories/)
+   - Separation: Controllers → Factories → ViewModels
+
+3. **Attempt Pattern** - Operations return `Attempt<TResult, TStatus>` for status-based error handling
+   - Controllers map status enums to HTTP status codes via helper methods
+
+4. **Authorization Service Pattern** - All authorization via `IAuthorizationService`, not attributes
+   - Checked in base controller methods (see `ManagementApiControllerBase`)
+
+5. **Options Pattern** - All configuration via `IOptions<T>` (security, routing, OpenAPI)
+
+6. **SignalR Event Broadcasting** - Real-time notifications via `BackofficeHub` and `ServerEventHub`
+
+---
+
+## 2. Commands
+
+### Build & Run
+```bash
+# Build
+dotnet build src/Umbraco.Cms.Api.Management
+
+# Test (tests in ../../tests/Umbraco.Tests.Integration and Umbraco.Tests.UnitTests)
+dotnet test --filter "FullyQualifiedName~Management"
+
+# Pack (for NuGet distribution)
+dotnet pack src/Umbraco.Cms.Api.Management -c Release
+```
+
+### Code Quality
+```bash
+# Format code
+dotnet format src/Umbraco.Cms.Api.Management
+
+# Build with warnings (note: some warnings suppressed, see .csproj line 23)
+dotnet build src/Umbraco.Cms.Api.Management /p:TreatWarningsAsErrors=true
+```
+
+### OpenAPI Documentation
+The project embeds a pre-generated `OpenApi.json` (1.3MB). To regenerate:
+```bash
+# Run Umbraco.Web.UI, access /umbraco/swagger
+# Export JSON from Swagger UI
+```
```

Comment:
Perhaps it would help to add the URL to the raw schema, which we (manually) copy over to the OpenAPI.json file: /umbraco/swagger/management/swagger.json